common/optionals.go

```diff
@@ -0,0 +1,13 @@
+package common
+
+// OptionalUint32 holds an optional uint32 value
+type OptionalUint32 struct {
```

Comment:
I these look similar to the optionals in elrond-go-core.
are these used with a different purpose or for elrond API calls.
Would it make sense to reference the ones in elrond-go-core if used for the API? 
maybe alias to the type in elrond-go-core?

Reply:
Right :)

I've updated the reference to `elrond-go-core` and used those definitions.